charts/helmet/templates/_helpers.tpl

```diff
@@ -8,3 +8,25 @@ Create the name of the service account to use
 {{- default "default" .Values.serviceAccount.name }}
 {{- end }}
 {{- end }}
+
+
+{{/*
+Render an array of env variables. The input can be a map or a slice.
+Values can be templates using the "common.tplvalues.render" helper, but changes to scope are not processed.
+Usage:
+{{ include "helmet.toEnvArray" ( dict "envVars" .Values.envVars "context" $ ) }}
+*/}}
+{{- define "helmet.toEnvArray" -}}
+{{- if kindIs "map" .envVars }}
+{{- range $key, $val := .envVars }}
+- name: {{ $key }}
+{{- if kindIs "string" $val }}
+  value: {{ include "common.tplvalues.render" (dict "value" $val "context" $.context) }}
+{{- else if kindIs "map" $val }}
+{{ include "common.tplvalues.render" (dict "value" (omit $val "name") "context" $.context) | indent 2 }}
```

Comment:
@zevisert Thanks for the changes. The only point I'd like to mention is that we want to support this kind of inputs for the `map` type:
```yaml
envVars:
  FOO: "foo"
  BAZ:
    valueFrom:
      configMapRef: ...
```
So you don't need to omit the "name" because it doesn't exist:
```suggestion
{{ include "common.tplvalues.render" (dict "value" $val "context" $.context) | indent 2 }}
```

Reply:
I was omitting the name here to prevent a case like this:

```yaml
envVars:
  FOO:
    name: BAR  # <-- Specifically what happens with this line
    valueFrom:
      configMapRef: ...
```

Since we just render the whole value (`$val`), there could be a conflict in names. Without the omit the above would be rendered as:

```yaml
spec:
  template:
    spec:
      containers:
        - env:
          - name: FOO
            name: BAR
            valueFrom: 
              configMapRef: ...
```

Sure, arguably, the additional name is user error, but it seems pretty easy to prevent that case by adding this omit